From 2afcd730a10d61c318289bc4c39c42eebd4b35a6 Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Mon, 29 Apr 2024 15:15:46 +0200
Subject: [PATCH 01/11] Prepare NEWS for branch 3.23 fixes

---
 NEWS | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/NEWS b/NEWS
index 11c67410fb6f..8bdd3753f7bb 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,14 @@
+Branch 3.23
+~~~~~~~~~~~
+
+* ==================== FIXED BUGS ====================
+
+The following bugs have been fixed or resolved on this branch.
+
+To see details of a given bug, visit
+  https://bugs.kde.org/show_bug.cgi?id=XXXXXX
+where XXXXXX is the bug number as listed above.
+
 Release 3.23.0 (26 Apr 2024)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-- 
2.45.2

